### ISSUE TITLE
How can I help with documentation?

### ISSUE THREAD

**Issue by pitsi** [justchokingaround/lobster#22]
Sorry for skipping the guidelines, but all 4 questions have nothing to do with what I ask for.
I noticed that lobster was updated to 3.0.1 a few hours ago (I actually updated it 20-30 minutes after it became available) and I also noticed the help request on the readme.
Although my knowledge is limited, I went through the script and through the api's settings and I really want to help with documentation. It is the least I can do as a thank you for your work!

So, what shall I include? How to select a server? What subtitle languages are supported and where? How to select the video quality when more than one is available?

**Comment by justchokingaround** [comment]
yes that is exactly what i want pls. if u could also pls include the locations of the default config and history file and what has changed since v2 that would be great, ty very much

**Comment by pitsi** [comment]
After trying out what the api can offer and what the script can do, I came accross 2 (and a half) issues.

One is mixdrop. The api does offer direct links to mixdrop streams, but they do not play at all. I even made a small oneliner that grabs the url, cleans it and and sends it to mpv, but mpv always returns a 403 error. I also tried setting mpv's user agent to some web browser! For that reason, setting mixdrop as the server in settings makes lobster return "no streams found" and exit, on any stream.
Do I mention it in the readme or do I omit it completely as a choice for servers and only mention the other 2?

The second one is resolution, or quality as it is mentioned. The api returns a quality named "auto", which is actually an hls stream that can adapt to the user's available bandwidth. Setting "auto" as the video_quality in settings has the same behavior as above, but this can be considered a bug on the script itself. I assume it can be patched, but it needs some more changes on how the script grabs the relevant url.
Like above, do I mention it or omit it?

And the "and a half" issue: subtitles. Not every stream has subtitles on every language. There are streams that have like 10 and there are streams that have english only. Shall I mention that as a warning or something?

**Comment by justchokingaround** [comment]
for how to make a pr:
1. fork the project
2. make a new branch for the documentation
3. make changes to that branch (you can do it in the browser if u want)
4. creat a pr

if u want u can just send me the md text and i'll deal with it myself.

for mixdrop, did u pass the referrer that the api gives back to mpv with ```--referrer "url"```? if mixdrop doesnt work then pls mention it in the docs as upcloud and vidcloud are pretty much the same thing. do mention the auto quality bandwith thing. and also pls mention the subs languages half issue

**Comment by pitsi** [comment]
About the pr... Yea, let me just send you the pm when I am done :)

For the rest.

You lost me on that referer thing, but I promise to look it up later. I had not noticed the referer part until now. However, shouldn't mixdrop work when added as a server in the settings?

If the auto quality is not supported and thus not working, why should I mention it? The way the script is right now, it just uses this scheme to play the relevant resolution that the api provides, i.e. one of those 3
```
https://example.com/blablabla/1080/index.m3u8
https://example.com/blablabla/720/index.m3u8
https://example.com/blablabla/360/index.m3u8
```
While the url for the auto (hls) stream looks like so
`https://example.com/blablabla/playlist.m3u8`

**Comment by pitsi** [comment]
Here is my first attempt. I have only documented configuration and... I do not know what to write for history.
I have omitted the mixdrop issue and the auto quality, since the first does not work and the second one is not implemented.

[lobsterstuff.txt](https://github.com/justchokingaround/lobster/files/9698613/lobsterstuff.txt)

**Comment by justchokingaround** [comment]
mixcloud doesnt work for me in the browser, even with a vpn so omitting mixdrop is a reasonable decision. i get a media not found error when running this 
```sh
curl "https://api.consumet.org/movies/flixhq/watch?episodeId=19651&mediaId=movie/watch-fight-club-19651&server=upcloud" 
```

**Comment by justchokingaround** [comment]
the documentation looks fine, i'll add everything else myself. i'll let u pr these changes to the readme, if u can figure out how to do that?

**Comment by pitsi** [comment]
> mixcloud doesnt work for me in the browser, even with a vpn so omitting mixdrop is a reasonable decision. i get a media not found error when running this
> 
> ```shell
> curl "https://api.consumet.org/movies/flixhq/watch?episodeId=19651&mediaId=movie/watch-fight-club-19651&server=upcloud" 
> ```

I do get results when running this. If I use mixdrop there, I also get "media not found". So I checked the regular page and the embeded player from mixdrop says "We can't find the video you are looking for". This means that the api is working correctly. Try another movie. I do all my tests with deadpool :D 

As for the text, just copy the contents from my file and paste them in your readme :)

**Comment by justchokingaround** [comment]
alright. i get the same results as you for curl and on the website. also i was meant to send mixdrop as server not upcloud in that command my bad

**Comment by pitsi** [comment]
URGENT!
Please correct this bit on line 2. This 
Settings are stored in ~/.config/lobster/lobster_config.txt
should read this
`Settings are stored in ~/.config/lobster/lobster_config.txt`

**Comment by justchokingaround** [comment]
ye i saw that dw, i always proof read

**Comment by pitsi** [comment]
Why is "auto" documented as an available quality when it does not work? That is why I omitted it! 
Proof
```
$ cat .config/lobster/lobster_config.txt 
player=mpv
subs_language=English
video_quality=auto
preferred_server=vidcloud
$ ./lobster.sh 
Enter a Movie/TV Show name: deadpool
Loading...
No links found
$
```
Proof with 720
```
$ cat .config/lobster/lobster_config.txt 
player=mpv
subs_language=English
video_quality=720
preferred_server=vidcloud
$ ./lobster.sh 
Enter a Movie/TV Show name: deadpool
Loading...
Press Enter to save movie progress or Ctrl-C to exit (this will not reset your progress)
^C
$
```
Notice how it immediately returns to the prompt when the "no links found" message shows up? With 720, the movie started playing after the loading message and I just exited it with q and pressed ctrl+c to return to the prompt.

It is because it tries to play a url that looks like this, only substituting the requested number with "auto"
`https://example.com/blablabla/auto/index.m3u8`
which is wrong because the actual url is this and has no "auto" in it.
`https://example.com/blablabla/playlist.m3u8`
That is why I said above that the auto quality is not supported yet. The script has a fixed way of selecting the stream and that is via the resolutions only. 
If you support auto, I will update the documented part and also add how much bandwidth is needed for each resolution.
